### PR TITLE
Move parameter logging to `get_integrator`

### PR DESCRIPTION
**Pull request by nefrathenrici**
From @Sbozzolo's comment on #2087

### CODE REVIEW

File: src/solver/types.jl [hunk 1]
```diff
@@ -459,6 +460,14 @@ function AtmosConfig(;
...
         @info "Running ClimaCore in unthreaded mode."
     end
+    if config["log_params"]
+        default_output = haskey(ENV, "CI") ? job_id : joinpath("output", job_id)
+        out_dir = parsed_args["output_dir"]
+        output_dir = isnothing(out_dir) ? default_output : out_dir
+        mkpath(output_dir)
+        logfilepath = joinpath(output_dir, "$(parsed_args["job_id"])_$FT.toml")
+        CP.log_parameter_information(toml_dict, logfilepath)
+    end
```
Comment (Sbozzolo) [inline src/solver/types.jl:470]: You could move this function to `get_integrator`, where the `output_dir` is already defined (after `get_simulation`)

File: src/parameters/create_parameters.jl [hunk 2]
```diff
@@ -7,9 +7,6 @@ import SurfaceFluxes.UniversalFunctions as UF
 import Insolation.Parameters as IP
 import Thermodynamics as TD
 import CloudMicrophysics as CM
-# TODO: Remove these imports?
-import ClimaCore
-import ClimaCore as CC
```
Comment (Sbozzolo) [inline src/parameters/create_parameters.jl:12]: What's the purpose of this change with regards to moving parameter logging?

Comment (nefrathenrici) [reply src/parameters/create_parameters.jl:12]: They don't seem to serve any purpose, I think they are old fragments that someone forgot to delete.

File: src/solver/types.jl (hunk 1 above)
Comment (nefrathenrici) [reply src/solver/types.jl]: TY!

File: src/parameters/create_parameters.jl (hunk 2 above)
Comment (Sbozzolo) [reply src/parameters/create_parameters.jl:12]: Maybe they should be removed in a different commit if they are not logically related to the change under consideration?

File: src/solver/type_getters.jl [hunk 3]
```diff
@@ -586,6 +586,11 @@ function get_simulation(config::AtmosConfig, comms_ctx)
     output_dir = isnothing(out_dir) ? default_output : out_dir
     mkpath(output_dir)
 
+    if parsed_args["log_params"]
+        logfilepath = joinpath(output_dir, "$(job_id)_$FT.toml")
+        CP.log_parameter_information(toml_dict, logfilepath)
+    end
+
```
Comment (Sbozzolo) [inline src/solver/type_getters.jl:593]: I would have put this directly in the `get_integrator`. I am not sure if logging the parameters logically belongs to to the `get_simulation` function. 

Also, maybe we can have the word `param`(eters) in the name of the output file

File: src/solver/type_getters.jl [hunk 4]
```diff
@@ -787,6 +787,11 @@ function get_integrator(config::AtmosConfig)
     atmos = get_atmos(config, params)
     numerics = get_numerics(config.parsed_args)
     simulation = get_simulation(config, config.comms_ctx)
+    simulation.output_dir
```
Comment (Sbozzolo) [inline src/solver/type_getters.jl:790]: I think you might have forgot to remove this line

File: src/solver/type_getters.jl [hunk 5]
```diff
@@ -787,6 +787,11 @@ function get_integrator(config::AtmosConfig)
...
+    if config.parsed_args["log_params"]
+        filepath = joinpath(output_dir, "$(job_id)_parameters.toml")
```
Comment (Sbozzolo) [inline src/solver/type_getters.jl:792]: I think this will fail because `output_dir` is not defined here.

File: src/solver/type_getters.jl (hunk 4 above)
Comment (nefrathenrici) [reply src/solver/type_getters.jl:790]: Yes, this is a typo. thanks!

**Review by Sbozzolo — APPROVED** [review]

**Review by charleskawczynski — APPROVED** [review]
